src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -46,8 +48,41 @@ public static IWebProxy DefaultProxy
             }
         }
 
-        public HttpRequestHeaders DefaultRequestHeaders =>
-            _defaultRequestHeaders ??= new HttpRequestHeaders();
+        /// <summary>Gets an <see cref="HttpClient"/> instance configured with all of the default options.</summary>
+        public static HttpClient Shared
+        {
+            get
+            {
+                return s_shared ?? Initialize();
+
+                static HttpClient Initialize()
+                {
+                    var shared = new HttpClient() { _isSharedInstance = true, _operationStarted = true };
+                    if (Interlocked.CompareExchange(ref s_shared, shared, null) is not null)
+                    {
+                        shared._isSharedInstance = false;
+                        shared.Dispose();
+                        shared = s_shared;
+                    }
+                    return shared;
+                }
+            }
+        }
+
+        public HttpRequestHeaders DefaultRequestHeaders
+        {
+            get
+            {
+                if (_isSharedInstance)
+                {
+                    // HttpRequestHeaders is not thread safe. Developers trying to mutate HttpClient.Shared.DefaultRequestHeaders
+                    // could thus corrupt the instance if used from multiple threads.
+                    throw new NotSupportedException(SR.net_http_shared_httpclient_notsupported);
```

Comment:
This makes me feel better about this feature.

Reply:
Not a fan of getter's throwing, I'd rather return a read-only object that threw if you tried to modify it.

Reply:
> I'd rather return a read-only object that threw if you tried to modify it.

That would a) make HttpRequestHeaders slower for everyone and b) not really affect anything because it has getters that mutate (e.g. lazily instantiate) and they would then need to throw instead.

Reply:
Don't we want to do the in `CancelPendingRequests()`, and any other member that has a chance to mess with independent callers?

Reply:
> Don't we want to do the same in CancelPendingRequests(), and any other member that has a chance to mess with independent callers?

We _could_.  CancelPendingRequests is "safe" to use concurrently, whereas DefaultRequestHeaders is not, hence why I gave the latter this treatment.